### PR TITLE
Add a simple flake measurement tool.

### PR DESCRIPTION
**Pull request by spxtr**
This looks at prow data for the last day and counts each time a test fails and passes for a single commit in a PR. It tends to undercount, but I think it's still useful. See the docstring for info.

Output right now:
```
Certain flakes from the last day:
9/143	(6%)	pull-kubernetes-e2e-gce-etcd3
9/132	(6%)	pull-kubernetes-e2e-gce-non-cri
5/145	(3%)	pull-kubernetes-node-e2e-non-cri
4/142	(2%)	pull-kubernetes-e2e-gce
4/145	(2%)	pull-kubernetes-unit
4/145	(2%)	pull-kubernetes-e2e-gke-gci
3/139	(2%)	pull-kubernetes-e2e-gce-gci
3/143	(2%)	pull-kubernetes-e2e-kops-aws
2/148	(1%)	pull-kubernetes-e2e-gke
2/40	(5%)	pull-kubernetes-cross
2/174	(1%)	pull-kubernetes-bazel
1/148	(0%)	pull-kubernetes-node-e2e
1/147	(0%)	pull-kubernetes-verify
1/142	(0%)	pull-kubernetes-kubemark-e2e-gce
Chance that a PR hits a flake: 32%
```

### CODE REVIEW

**Comment by rmmh** [comment]
/lgtm

this would be a nice daily periodic job :-)

File: experiment/flakedetector.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+isn't a perfect signal, since something might have happened on master that
+makes it flake, but I think it's good enough.
+
+There will also be false negatives: flakes that don't show up here because the
+PR author changed the PR. Still, this is a good signal.
+"""
+
+import json
+import operator
+import requests
```
Comment (rmmh) [inline experiment/flakedetector.py]: linebreak between system imports and third-party imports (requests)

Comment (spxtr) [inline experiment/flakedetector.py]: Done.

**Comment by spxtr** [comment]
Added flake chance calculation and updated the docstring.

File: experiment/flakedetector.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        if 'success' in results and 'failure' in results:
+            job_flakes[job] += 1
+
+print('Certain flakes from the last day:')
+total_success_chance = 1.0
+for job, flakes in sorted(job_flakes.items(), key=operator.itemgetter(1), reverse=True):
+    fail_chance = flakes / job_commits[job]
+    total_success_chance *= (1.0 - fail_chance)
+    print('{}/{}\t({}%)\t{}'.format(flakes, job_commits[job], int(100*fail_chance), job))
+print('Chance that a PR hits a flake: {}%'.format(int(100*(1-total_success_chance))))
```
Comment (rmmh) [inline experiment/flakedetector.py]: `{:.0f}` would remove the int() call

Comment (spxtr) [inline experiment/flakedetector.py]: Done.

File: experiment/flakedetector.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+A flake is counted if a job passes and fails for the same pull commit. This
+isn't a perfect signal, since something might have happened on master that
+makes it flake, but I think it's good enough.
+
+There will also be false negatives: flakes that don't show up here because the
+PR author changed the PR. Still, this is a good signal.
+
+The chance of hitting a flake is calculated assuming they are uncorrelated,
+which should be a pretty good assumption unless we have a short outage. For
+instance, if all tests start to fail for just a few minutes, then a whole
```
Comment (kargakis) [inline experiment/flakedetector.py]: It seems we should be able to identify these outages.

**Pull request merged**